articles/building-your-web-server-with-vlang/index.md

```diff
@@ -0,0 +1,264 @@
+### Getting Started with V
+
+Let's proceed to installing V and writing our first V program.
+
+#### Prerequesites
+
+- [Git](https://git-scm.com/download/win)
+- A C compiler like [tcc](https://developerinsider.co/download-turbo-c-for-windows-7-8-8-1-and-windows-10-32-64-bit-full-screen/) (preferably)
+- Terminal (cmd, powershell or bash)
+- Code Editor like VSCode (preferably)
+
+#### Installation
+
+For Linux, macOS, FreeBSD, etc:
+
+```bash
+git clone https://github.com/vlang/v
+cd v
+make
+sudo ./v symlink 
+```
+
+For Windows:
+
+```powershell
+git clone https://github.com/vlang/v
+cd v
+make.bat
+.\v.exe symlink # this adds V to the PATH of your environmental variables
+```
+
+When your installation is complete, reopen the terminal, then run this command to confirm that V is installed.
+
+ 
+
+```bash
+v version
+```
+
+You should get `V 0.2.2 2b53992` . 
+
+#### Writing your first V program
+
+Open your code editor and if you use VSCode, install `vlang-vscode.extension`. 
+
+The extension gives you syntax highlighting, linting, formatting, and code snippet support for Vlang. 
+
+Let's now write our first Hello World program with V.
+
+Run this command to create a `hello.v` file.
+
+```bash
+touch hello.v
+```
+
+Then add the following lines of code to the file 
+
+```c
+//hello.v
+fn main() {
+  println('Hello V World!')
+}
+```
+
+Run `v run hello.v` on your terminal to get the output.
+
+You should get `Hello V World!` 
+
+### Using V on the web
+
+The official V documentation claims that anything you can do in other languages, you can do in V.
+
+The V programming language has a built-in library for building web applications called **vweb. They built vweb from the scratch with V; it is tiny and easy to deploy. It’s still in the pre-alpha stage.
+
+Notwithstanding, let's go-ahead to write a simple web server with vweb. 
+
+#### Building a web server with V
+
+Create a `server.v` file and add the following lines of code.
+
+```c
+//server.v
+**import vweb
+
+struct App {
+    vweb.Context
+}
+
+fn main() {
+    vweb.run<App>(8080)
+}
+
+pub fn (mut app App) index() vweb.Result {
+	return app.text('Hello world from vweb!')
+}**
+```
+
+First, import the `vweb` module.
+
+Next, you create a struct to hold any variable that you'll need in the app. For instance, you could have something like this:
+
+```c
+struct App{
+	mut:
+		count int
+}
+```
+
+and then make use of `count` somewhere else in the program.
+
+Next, you start the webserver and listen to a port using`fn run<T>(port int)` . 
+
+Finally, to create routes in V, you can either specify the path as an attribute like this `["/"]` or auto-mapping of function names like this `index()` .
+
+ Compile the program using the v compiler, run `v run server.v` on your terminal.
+
+You should get something like this on your terminal:
+
+![Building%20your%20Web%20Server%20with%20Vlang%2055ee48a335b145df9af2fe6d5f312905/Untitled.png](/articles/building-your-web-server-with-vlang/v-server.png)
+
```

Comment:
Please format your image to look like this: 
```markdown
![vlang server](/engineering-education/building-your-web-server-with-vlang/v-server.png)
```
Please format all your images as shown above.